Update classification_and_policy_enforcement.adoc (#1261)
owncloud/docs-server: modules/admin_manual/pages/enterprise/document_classification/classification_and_policy_enforcement.adoc

Administrators can find examples and generalized configuration instructions below.

=== EXIF Header Metadata

> PDF and several Image formats (JPEG, PNG, HEIC) support EXIF metadata. For a full list of supported file formats see the documentation of exiftool.
> Support for EXIF was added in version 1.5.0 of the files_classifier app. Please note, that the feature requires the installation of exiftool. (E.g. 
> sudo apt install exiftool)

Reviewer: ```suggestion

IMPORTANT: For several file types, the classification relies on the installation of `exiftool`. For details see below.

PDF and several Image formats like JPEG, PNG, HEIC support EXIF metadata. See the documentation for a full list of https://exiftool.org/#supported[supported file formats] of exiftool. Support for EXIF was added in version 1.5.0 of the {oc-marketplace-url}/app/files_classifier[Files Classifier app]. Please note that the feature requires the installation of exiftool. 

[source,bash]
----
sudo apt install exiftool
----
```

Administrators can find examples and generalized configuration instructions below.

=== EXIF Header Metadata

> 

Reviewer: ```suggestion
```

** Hit btn:[Save].
* To verify that the rule is in place, upload a classified file, check for the tag and try to create a public link share.

===== EXIF Metadata

> The available elements in EXIF metadata can vary depending how the file was created.

Reviewer: ```suggestion
The available elements in EXIF metadata can vary depending how the file was created.

```

** Hit btn:[Save].
* To verify that the rule is in place, upload a classified file, check for the tag and try to create a public link share.

===== EXIF Metadata

> The available elements in EXIF metadata can vary depending how the file was created.
> * Photo cameras may embed the Product name: **Property XPath** = `//IFD0:Model` and e.g. **Property Value** = `Canon EOS 40D`
> * Software creating PDF documents may embed their name: 
>   ** Property XPath** = `//xmp:CreatorTool` and **Property Value** = `Adobe InDesign 15.0 (Macintosh)`
> * Screenshot or graphics software may have e.g.
>   ** Property XPath** = `///PNG:Software` and **Property Value** = `gnome-screenhot`
>   ** Property XPath** = `//PNG:Comment` and **Property Value** = `Created with GIMP`

Reviewer: ```suggestion
The available elements in EXIF metadata can vary depending how the file was created.
* Software creating PDF documents may embed their name: +
**Property XPath** = `//xmp:CreatorTool` and **Property Value** = `Adobe InDesign 15.0 (Macintosh)`

* Photo cameras may embed the Product name: +
**Property XPath** = `//IFD0:Model` and e.g. **Property Value** = `Canon EOS 40D`

* Screenshot or graphics software may have: +
**Property XPath** = `///PNG:Software` and **Property Value** = `gnome-screenhot` +
**Property XPath** = `//PNG:Comment` and **Property Value** = `Created with GIMP`
```

- Note down the classification property and value.
- Repeat the steps for all classification properties and values you want to set up classification rules for in ownCloud.

==== Finding Metadata in PDF or Images

- The command line tool `exiftool` can be used to inspect a file for suitable metadata. 

> - The output of `exiftool -n -q -b -X example.pdf` shows the Metadta XML-Syntax. Use the XML node names to generate XPath queries and XML node data for the Property Value.

Reviewer 2: ```suggestion
- The output of `exiftool -n -q -b -X example.pdf` shows the Metadata XML-Syntax. Use the XML node names to generate XPath queries and XML node data for the Property Value.
```

- Note down the classification property and value.
- Repeat the steps for all classification properties and values you want to set up classification rules for in ownCloud.

==== Finding Metadata in PDF or Images

> - The command line tool `exiftool` can be used to inspect a file for suitable metadata. 
> - The output of `exiftool -n -q -b -X example.pdf` shows the Metadata XML-Syntax. Use the XML node names to generate XPath queries and XML node data for the Property Value.
> - The metadata may use XML Namespaces. In `xmp:CreatorTool` the namespace is `xmp`. 
> - We can use some more complex XPath expression to match the CreatorTool e.g. in any namespace or by prefix:
>   ** **Property XPath** = `//*[local-name()='Comment']` and **Property Value** = `Created with GIMP`
>   ** **Property XPath** = `//*[local-name() = 'CreatorTool']`
>   ** **Property XPath** = `//*[starts-with(local-name(), 'Creator')]`
> 

Reviewer: ```suggestion
* The command line tool `exiftool` can be used to inspect a file for suitable metadata. 
* The output of `exiftool -n -q -b -X example.pdf` shows the metadata XML syntax. Use the XML element names to generate XPath queries and XML element data for the Property Value.
* The metadata may use XML namespaces. In `xmp:CreatorTool` the namespace is `xmp`. 
* We can use more complex XPath expressions to match the `CreatorTool` like in any namespace or by prefix: +
**Property XPath** = `//*[local-name()='Comment']` and **Property Value** = `Created with GIMP` +
**Property XPath** = `//*[local-name() = 'CreatorTool']` +
**Property XPath** = `//*[starts-with(local-name(), 'Creator')]`

Note that other XPath selectors than `local-name()` or `starts-with` are currently not supported.
```
